lib/submit-request.js

```diff
@@ -185,6 +185,39 @@ SubmitRequest.prototype.commit = function(callback) {
   });
 };
 
+SubmitRequest.prototype._metadataProjection = function() {
+  var request = this;
+
+  // Default behavior
+  if (!request.opMetadataProjection) {
+    return undefined;
+  }
+
+  // Granular projection
+  if (typeof request.opMetadataProjection === 'object') {
+    return request._granularMetadataProjection();
+  }
+
+  // Full projection
+  return request.op.m;
+};
+
+SubmitRequest.prototype._granularMetadataProjection = function() {
+  var request = this;
+  var metadataProjection = {};
+  for (var key in request.opMetadataProjection) {
+    var doProject = request.opMetadataProjection[key];
+    if (! doProject ) {
+      continue;
+    }
+
+    metadataProjection[key] = request.op.m[key];
```

Comment:
You can make this a bit more readable by inverting the logic here:

```suggestion
    if (doProject) {
      metadataProjection[key] = request.op.m[key];
    }
```